parley/src/layout/line/greedy.rs

```diff
@@ -602,16 +605,53 @@ impl<'a, B: Brush> BreakLines<'a, B> {
                 }
             }
 
+            if quantize {
+                // We round line height to full pixels and then keep track of the overflow.
+                let rounded_line_height = line.metrics.line_height.round();
+                let overflow = line.metrics.line_height - rounded_line_height;
+                line.metrics.line_height = rounded_line_height;
+                // The overflow tracking logic is based on observing Chrome's behavior.
+                // Fractional line height will lead to fluctuating integral perceived line height.
+                if y_overflow <= -0.5 {
+                    // This line will overflow to the previous line.
+                    y -= 1.;
+                    y_overflow += 1.;
+                } else if y_overflow >= 0.5 {
+                    // There will be a 1px gap before the start of the line.
+                    y += 1.;
+                    y_overflow -= 1.;
+                }
+                y_overflow += overflow;
+                // We also mimic Chrome in rounding ascent/descent.
+                line.metrics.ascent = line.metrics.ascent.round();
+                line.metrics.descent = line.metrics.descent.round();
+            }
             line.metrics.leading =
                 line.metrics.line_height - (line.metrics.ascent + line.metrics.descent);
-
+            // We mimic Chrome in giving 'above' the remainder in cases of odd leading.
+            let leading_below = if quantize {
+                (line.metrics.leading * 0.5).trunc()
+            } else {
+                line.metrics.leading * 0.5
+            };
+            let leading_above = if quantize {
+                line.metrics.leading - leading_below
+            } else {
+                line.metrics.leading * 0.5
+            };
```

Comment:
Nit: I think it would read clearer if these `if-else` blocks were merged.

Reply:
You mean like this?

```rust
let (leading_above, leading_below) = if quantize {
    let below = (line.metrics.leading * 0.5).trunc();
    let above = line.metrics.leading - below;
    (above, below)
} else {
    let half = line.metrics.leading * 0.5;
    (half, half)
};
```

Reply:
Yes, although I hadn't noticed the blocks were returning a value, which makes it not quite as neat.

Another possibility which might be nice is:

```rust
let leading_below = if quantize {
       (line.metrics.leading * 0.5).trunc()
} else {
       line.metrics.leading * 0.5
};
let leading_above = line.metrics.leading - leading_below;
```

I'll leave it up to your judgement (the original version is also probably fine).